build-tools/mono-runtimes/mono-runtimes.targets

```diff
@@ -72,6 +72,70 @@
         WorkingDirectory="$(MonoSourceFullPath)"
     />
   </Target>
+
+  <Target Name="_PrepareLlvmItems">
+    <ItemGroup>
+      <_LlvmSourceFile Include="$(LlvmSourceFullPath)\lib\**\*.cpp" />
+      <_LlvmHeaderFile Include="$(LlvmSourceFullPath)\lib\**\*.h" />
+      <_LlvmMakefileConfig Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Makefile.config')" />
+      <_LlvmConfigureStamp Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\.stamp-configure')" />
+    </ItemGroup>
+
+    <ItemGroup>
+      <_LlvmArchive Include="$(_LlvmOutputDirTop)\%(_LlvmRuntime.BuildDir)\Release\lib\*.a" />
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\opt%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' "/>
+
+      <_LlvmSourceBinary Include="@(_LlvmRuntime->'$(_LlvmOutputDirTop)\%(BuildDir)\Release\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+      <_LlvmTargetBinary Include="@(_LlvmRuntime->'$(OutputPath)\bin\llc%(ExeSuffix)')" Condition=" '%(_LlvmRuntime.InstallBinaries)' == 'true' " />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ConfigureLlvm"
+          DependsOnTargets="_PrepareLlvmItems"
+          Inputs="$(LlvmSourceFullPath)\Makefile.config.in;$(LlvmSourceFullPath)\configure;@(_LlvmSourceFile);@(_LlvmHeaderFile)"
```

Comment:
`_ConfigureLlvm` depends on all the source and header files of llvm, however. Any update to any of them will trigger a rebuild.